serverless/src/getConcepts/__tests__/handler.test.js

```diff
@@ -305,90 +319,134 @@ describe('getConcepts', () => {
-
-          const conceptMatches = result.body.match(/<skos:Concept/g) || []
-          expect(conceptMatches.length).toBe(expectedConceptCount)
-
-          // Check that the correct range of concepts is included
-          const startIndex = (parseInt(pageNum, 10) - 1) * parseInt(pageSize, 10)
-          const endIndex = Math.min(startIndex + parseInt(pageSize, 10), 100)
-          const expectedConcepts = Array.from(
-            { length: endIndex - startIndex },
-            (_, i) => i + startIndex
-          )
-
-          expectedConcepts.forEach((i) => {
-            expect(result.body).toContain(`<skos:Concept rdf:about="uri${i}">`)
-            expect(result.body).toContain(`<skos:prefLabel>Concept uri${i}</skos:prefLabel>`)
-          })
-
-          expect(getGcmdMetadata).toHaveBeenCalledWith({
-            pageNum: parseInt(pageNum, 10),
-            pageSize: parseInt(pageSize, 10),
-            gcmdHits: 100,
-            version: 'published'
-          })
-        }))
+      getFilteredTriples.mockResolvedValue(allMockTriples.slice(25, 50))
+
+      const result = await getConcepts(event)
+
+      expect(result.headers['X-Total-Count']).toBe('100')
+      expect(result.headers['X-Page-Number']).toBe('2')
+      expect(result.headers['X-Page-Size']).toBe('25')
+      expect(result.headers['X-Total-Pages']).toBe('4')
+
+      const conceptMatches = result.body.match(/<skos:Concept/g) || []
+      expect(conceptMatches.length).toBe(25)
+
+      for (let i = 25; i < 50; i += 1) {
+        expect(result.body).toContain(`<skos:Concept rdf:about="uri${i}">`)
+        expect(result.body).toContain(`<skos:prefLabel>Concept ${i}</skos:prefLabel>`)
+      }
+
+      expect(getFilteredTriples).toHaveBeenCalledWith({
+        conceptScheme: undefined,
+        pattern: undefined,
+        version: 'published',
+        pageNum: 2,
+        pageSize: 25
+      })
+    })
+
+    it('handles pagination for last page with fewer items', async () => {
+      const event = {
+        queryStringParameters: {
+          page_num: '4',
+          page_size: '30',
+          format: 'rdf'
+        }
+      }
+
+      getFilteredTriples.mockResolvedValue(allMockTriples.slice(90, 100))
+
+      const result = await getConcepts(event)
+
+      expect(result.headers['X-Total-Count']).toBe('100')
+      expect(result.headers['X-Page-Number']).toBe('4')
+      expect(result.headers['X-Page-Size']).toBe('30')
+      expect(result.headers['X-Total-Pages']).toBe('4')
+
+      const conceptMatches = result.body.match(/<skos:Concept/g) || []
+      expect(conceptMatches.length).toBe(10)
+
+      for (let i = 90; i < 100; i += 1) {
+        expect(result.body).toContain(`<skos:Concept rdf:about="uri${i}">`)
+        expect(result.body).toContain(`<skos:prefLabel>Concept ${i}</skos:prefLabel>`)
+      }
+
+      expect(getFilteredTriples).toHaveBeenCalledWith({
+        conceptScheme: undefined,
+        pattern: undefined,
```

Comment:
Do we need a test where these patterns are defined I guess thats really just going to be handled on the db side?